src/main.ts

```diff
@@ -229,11 +230,13 @@ export class Purchases {
    * @param rcPackage - The package you want to purchase. Obtained from {@link Purchases.getOfferings}.
    * @param customerEmail - The email of the user. If null, RevenueCat will ask the customer for their email.
    * @param htmlTarget - The HTML element where the billing view should be added. If null, a new div will be created at the root of the page and appended to the body.
+   * @param purchaseOptionId - The ID for a specific purchase option. Used to purchase trials and other offers. Obtained from {@link Purchases.getOfferings}.
    * @returns The customer info after the purchase is completed successfuly.
    * @throws {@link PurchasesError} if there is an error while performing the purchase. If the {@link PurchasesError.errorCode} is {@link ErrorCode.UserCancelledError}, the user cancelled the purchase.
    */
   public purchasePackage(
     rcPackage: Package,
+    purchaseOptionId: string = "base_option",
```

Comment:
This is a breaking change.

@tonidero I think we might want to handle these parameters with an object to avoid future changes to cause a breaking change.

Something like:
```
public purchasePackage(
    rcPackage: Package,
    options: {
        purchaseOptionId?: string | undefined,
        customerEmail?: string,
        htmlTarget?: HTMLElement        
    }
```

So that clients will be able to use it the fields as named parameters instead of ordered ones

```
 purchases.purchasePackage(package, { purchaseOptionId: 'some_id', htmlTarget: element}
```

Reply:
Yes! I agree with that approach. One thing I would do for now though is to keep the old one and deprecate it and add this one as a separate method. Also, we can give a name to the options like `PurchaseOptions`  or `PurchaseParams`

Reply:
What about `purchaseSubscriptionOption` for the name of the method?

Regarding the type for the params: yes I'll create an interface for it and call it `PurchaseOptions`.

I'll fix another couple of bugs and add some tests and then I'll change this interface, thanks for answering to the comment!

Reply:
> What about purchaseSubscriptionOption for the name of the method?

Hmm I think I would keep it as `purchasePackage` if possible? Just with different parameters. That way we try to keep everything into a single purchase method. 

In the android SDK we had a similar situation and had a ton of purchase methods. In the end, we switched to the builder pattern and now have a single purchase method that receives an object constructed from `PurchaseParams.Builder`. I think that exact approach might not be as common in TS, but having an object for all optional parameters makes the most sense to me.

Reply:
Since in typescript overloads allow only one implementation:
https://www.typescriptlang.org/docs/handbook/2/functions.html#function-overloads

I would prefer to define the `purchase` method and add only the `package` parameter as positional (and mandatory).

For all the rest we will use the params object.

Wdyt?

Reply:
I updated the code with the `purchase` method here: https://github.com/RevenueCat/purchases-js/pull/117/files#diff-4fab5baaca5c14d2de62d8d2fceef376ddddcc8e9509d86cfa5643f51b89ce3dR260

Reply:
Hmm ok, the idea of naming it `purchasePackage` originally is that this will purchase a package, and we might want to allow someone not to use offerings in the future and purchase a product directly.

WDYT about adding the package to the PurchaseOptions type as mandatory instead? That way in the future we can try to make that object accept either a package or a product as mandatory and process it internally.

Reply:
I am fine with that, I can migrate all the parameters to the object and that's it.

Reply:
Done!